File: java/src/tablelize/Row.java
package tablelize;

import java.util.Hashtable;
import java.util.Map;

public class Row {
	
	private Map<String, String> row = new Hashtable<String, String>();
	
	public void addField(String fieldName, String fieldValue) {
		this.row.put(fieldName, fieldValue);
	}
	
	public String getFieldValue(String fieldName) throws Exception {
		if (!this.row.containsKey(fieldName))
			throw new Exception("cant find a column with name: " + fieldName);
			
		return this.row.get(fieldName);
	}
	
	public int getFieldsCount() {
		return this.row.size();
	}

}

File: csharp/TablelizeIt/Row.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TablelizeIt
{
    public class Row
    {
        private Dictionary<string, string> row = new Dictionary<string, string>();
	
        public void AddField(string fieldName, string fieldValue) 
        {
            this.row.Add(fieldName, fieldValue);
        }
	
        public string GetFieldValue(string fieldName) 
        {
            if (!this.row.ContainsKey(fieldName))
                throw new Exception("cant find a column with name: " + fieldName);
                
            return this.row[fieldName];
        }
	
        public int GetFieldsCount() 
        {
            return this.row.Count;
        }
    }
}
